mathics/builtin/assignments/internals.py

```diff
@@ -567,9 +567,12 @@ def process_assign_makeboxes(self, lhs, rhs, evaluation, tags, upset):
     #   MakeBoxes[CubeRoot, StandardForm] := RadicalBox[3, StandardForm]
     # rather than:
     #   MakeBoxes[CubeRoot, StandardForm] -> RadicalBox[3, StandardForm]
-    makeboxes_rule = Rule(lhs, rhs, system=True)
-    makeboxes_defs = evaluation.definitions.builtin["System`MakeBoxes"]
-    makeboxes_defs.add_rule(makeboxes_rule)
+
+    makeboxes_rule = Rule(lhs, rhs, system=False)
+    definitions = evaluation.definitions
+    definitions.add_rule("System`MakeBoxes", makeboxes_rule, "down")
```

Comment:
Is the reason, "add_rule" takes a string here a historical artifact, or is there a reason why a string is better than a Symbol?  Semantically, Symbol is the more precise entity, right?

Reply:
Symbol would be more precise, but now, the key in  `Definitions` is a Python `str`. To change it by a Symbol is not trivial (I tried once). So, the answer is both: historical and efficiency reasons.